libs/pika/runtime/src/runtime.cpp

```diff
@@ -939,17 +939,17 @@ namespace pika::detail {
                     !pika::threads::detail::any(pu_mask) ? ~pu_mask : pu_mask;
 
                 // The masks reported by pika must be the same as the ones
-                // reported from HWLOC.
+                // reported from hwloc.
                 if (!ec &&
                     !pika::threads::detail::equal(
                         boundcpu_never_empty, pu_mask_never_empty, num_threads))
                 {
                     std::string boundcpu_str =
                         pika::threads::detail::to_string(boundcpu_never_empty);
                     std::string pu_mask_str = pika::threads::detail::to_string(pu_mask_never_empty);
-                    PIKA_THROW_EXCEPTION(pika::error::invalid_status, "handle_print_bind",
-                        "unexpected mismatch between binding reported from HWLOC({}) and "
-                        "pika({}) on thread {}",
+                    PIKA_LOG(warn,
+                        "unexpected mismatch between binding reported from hwloc ({}) and "
+                        "binding expected by pika ({}) on thread {}",
```

Comment:
Perhaps mention `--pika:print-bind` here to check the bindings?

Reply:
@RMeli what do you think about https://github.com/pika-org/pika/pull/1251/commits/11314692fa49d0fc934640e59e20cb26eae2c4bf?

Reply:
Looks great, thanks!